lib/param_parsing/terragrunt_test.go

```diff
@@ -0,0 +1,19 @@
+package param_parsing
+
+import (
+	"github.com/hashicorp/go-version"
+	"testing"
+)
+
+func TestGetVersionFromTerragrunt(t *testing.T) {
```

Comment:
Could there be a test for checking the decode failures for the `diagnostics.HasErrors()` branch?

Reply:
This would require refactoring that the method returns an error instead of a logger.Fatal, which causes an os.exit(1)
It might be the cleaner approach though. 🤔

Reply:
Added test-cases for that